charts/tidb-cluster/templates/tidb-initializer-job.yaml

```diff
@@ -23,22 +24,50 @@ spec:
         image: {{ .Values.mysqlClient.image }}
         imagePullPolicy: {{ .Values.mysqlClient.imagePullPolicy | default "IfNotPresent" }}
         command:
-        - /bin/sh
+        - python
         - -c
-        # Read sql from file to avoid special characters interpreted as builtin variable
-        # And also avoid plain text password show in Job and Pod spec
-        # Besides we can also add more SQL in the file for initialization, eg. create database, create user etc
         - |
-          until mysql -h {{ .Values.clusterName }}-tidb -P 4000 --connect-timeout=5 < /data/init-password.sql; do sleep 2; done
+          import os, MySQLdb
+          host = {{ printf "%s-tidb" .Values.clusterName | quote }}
+          port = 4000
+          root_password = os.environ.get('ROOT_PASSWORD')
+          system_password = os.environ.get('SYSTEM_PASSWORD')
+          conn = MySQLdb.connect(host=host, port=port, user='root', connect_timeout=5)
+          if root_password:
+              conn.cursor().execute("set password for 'root'@'%%' = %s;", (root_password,))
+          if system_password:
+              conn.cursor().execute("create user 'system'@'%%' identified by %s;", (system_password,))
+          conn.cursor().execute("flush privileges;")
+          conn.commit()
+          {{- if .Values.tidb.initSql }}
+          with open('/data/init.sql', 'r') as sql:
+              for line in sql.readlines():
+                  conn.cursor().execute(line)
+                  conn.commit()
+          {{- end }}
+        env:
+        - name: ROOT_PASSWORD
+          valueFrom:
+            secretKeyRef:
+              name: {{ .Values.tidb.passwordSecret }}
+              key: root
+        - name: SYSTEM_PASSWORD
```

Comment:
The setup here of the secret being a `user: password` mapping is nice. Users would not need to modify this yaml file if we could loop over the `user: password` pairs in the Python code to set them. Instead they just set the secret user-password pairs. I think this should be possible by mounting the secret as a file.

Reply:
When there are multiple data in a secret, the secret volume is mounted as multiple files. But this is possible, we can mount secret as `/etc/tidb/secrets`, and having secret data named as `<user>_password`, then each user's password is storing in `/etc/tidb/secrets/<user>_password`. And then looping the files in Python script.

Reply:
Yes! I would just have it as `<user>` rather than `<user>_password. The secret is just for TiDB passwords.